lib/counting.hh

```diff
@@ -202,6 +213,51 @@ public:
 
     } // count
 
+    virtual void count_ts(const char * kmer)
+    {
+        HashIntoType hash = _hash(kmer, _ksize);
+        count_ts(hash);
+    }
+
+    // Thread-safe counting function
+    virtual void count_ts(HashIntoType khash)
+    {
+        bool is_new_kmer = true;
+        unsigned int  n_full      = 0;
+        uint32_t lock_index = khash % _n_table_locks;
+        while(!__sync_bool_compare_and_swap( &_table_spinlocks[lock_index], 0, 1));
```

Comment:
for readability, maybe put a comment (or make a macro) that says "take lock", and "release lock" (below).